src/components/containers/PanelHeader.js

```diff
@@ -74,13 +55,18 @@ class PanelHeader extends Component {
   }
 }
 
+PanelHeader.contextTypes = {
+  layout: PropTypes.object,
+  onUpdate: PropTypes.func,
+  updateContainer: PropTypes.func,
+};
+
 PanelHeader.propTypes = {
-  action: PropTypes.string,
+  addAction: PropTypes.object,
   allowCollapse: PropTypes.bool,
```

Comment:
this is really nice! but should this 'action' stay more general, maybe it's not always going to be an 'addAction'

Reply:
At the moment it's pretty well coupled to being `add` if we hardcode the `<PlusIcon>` like we do... Should I generalize the icon as well?

Reply:
yeah, you're right, I didn't notice the icon, we should be generalizing that too then if going that route.

Well, ok, let's generalize when we need to generalize. I don't foresee us needing other actions in the panel header, other than add, anytime soon